CorsixTH/Lua/dialogs/build_room.lua

```diff
@@ -102,16 +102,19 @@ function UIBuildRoom:UIBuildRoom(ui)
   self:makeTooltip(_S.tooltip.build_room_window.cost, 160, 228, 282, 242)
 end
 
+--! Checks what rooms are now available to build
 function UIBuildRoom:updateBuildableRooms()
   local app = self.ui.app
   self.category_rooms = {}
   for i, category in ipairs({"diagnosis", "treatment", "clinics", "facilities"}) do
     local rooms = {}
     self.category_rooms[i] = rooms
-    for _, room in ipairs(app.world.available_rooms) do
+    local room_discoveries = self.ui.hospital.room_discoveries
```

Comment:
The right-hand side doesn't depend on `i`, `category`, `rooms`, or `self.category_rooms[i]`, this statement here does 4 times the same thing. Perhaps move it outside the loop?

Reply:
If you fix the thing below, you may find there is only one use left for `room_discoveries`, at which point it may be beneficial to completely remove this variable, and copy  the right-hand side at that spot instead.